### ISSUE TITLE
Reproducible - Google Cloud Ubuntu 20.04 - v1.6.5 - Memory growth until system locks up with K3s

### ISSUE THREAD

**Issue by danmanners** [zerotier/ZeroTierOne#1423]
TL;DR - When I spin up an Ubuntu 20.04 server in Google Cloud (and **ONLY** in Google Cloud), after installing (and configuring) K3s and ZeroTier, ZeroTier begins to grow in memory utilization until it crashes the system.

# Expected

- When I initialize a Google Cloud e2-small (2vCPU, 2GiB Memory), I am installing ZeroTier (through the `curl` command), K3s (using ~100-300MiB Memory on average), and NGINX (running TCP Proxy).
- Zerotier starts first, then K3s, and finally NGINX.
- K3s is using the Zerotier interface as it's flannel-iface setting (example: `flannel-iface: ztyou2j6dw`)
- Expected behavior is that things work for longer than 3-5 minutes total.

# Unexepcted

- Over 1-3 minutes, the ZeroTier memory utilization begins to climb until it crashes the local system, requiring a hard reset.

This is only repeatable in Google Cloud. I am running the exact same size VMs in AWS and Digital Ocean as well, and cannot repeat these issues.

# Repeating the steps

- This is consistently repeatable, and I'm happy to show or explain more.

# Additional information 

## Screenshots

![image](https://user-images.githubusercontent.com/17593824/123638720-7d0a9a80-d7ed-11eb-91e9-f8e0905edc83.png)

## Notes & Additional Context

**Desktop (please complete the following information):**
 - OS: Ubuntu Linux 20/04
 - Environment: Google Cloud
 - ZeroTier Version: 1.6.5
  
**Additional context**
- All expected ports are available, and until zerotier runs out of memory, for ~1-3 minutes, everything works fine and completely as expected.

**Comment by glimberg** [comment]
Do you have instructions for a minimal environment for reproduction?

**Comment by danmanners** [comment]
@glimberg I've got a bunch of scratchpad notes [here](https://github.com/danmanners/homelab-k3s-cluster/tree/main/Terraform) for getting my cloud infrastructure deployed in both Digital Ocean **and** Google Cloud, and wrote [this blog post](https://www.danmanners.com/posts/p2-k3s-digitalocean-zerotier-and-more/) covering the steps I took getting my homelab ZeroTier router up and going, as well as getting cloud hosts connected back home. Trying to figure out exactly how to provide good **minimal** instructions for getting everything up and going outside of my homelab/cloud infrastructure now though.

Also more than happy to add a public key to a new host in Google Cloud if that would be easier. Happy to put whatever together.

## K3s & Zerotier Networking Diagram

![image](https://user-images.githubusercontent.com/17593824/123678070-d7b6ed00-d813-11eb-9c0b-5e81d7acd0c1.png)

**Comment by glimberg** [comment]
Yeah I'd found a lot of that already.  That's why I was asking for a minimal case that demonstrates the issue.

**Comment by someara** [comment]
Thanks for this, @danmanners

**Comment by danmanners** [comment]
@glimberg / @laduke, I will verify one or two more things tonight and put together exact steps to replicate. May put it all in a repo for easy replication. Sorry I didn't have this already ready!

**Comment by danmanners** [comment]
@glimberg / @laduke, still need to verify everything and it's getting late here, but [here's the repo I've put together](https://github.com/danmanners/zerotier-ubuntu-google-troubleshooting). This **should** be more than enough to replicate the problem.

Again, haven't 100% verified everything here, but will do this tomorrow and throw a thumbs-up if this successfully replicates the issues.

**Comment by danmanners** [comment]
Hoping to have an RCA put together, but heads up my Google Account was just suspended for "cryptomining" while running nothing but K3s and Zerotier while watching the memory leak continue to grow slowly. Should have more information available in the next 48 hours, depending on Google Cloud's security team.

**Comment by glimberg** [comment]
Yeah. Pinged you there on Twitter.  I'm now pretty certain that you're hitting a feedback loop where for some reason the OS is routing ZeroTier packets back through ZeroTier.  The issue is pretty well documented over here: https://github.com/zerotier/ZeroTierOne/issues/779.  Might have to do some packet sniffing to figure out which address blocks are being routed back into the feedback loop.

**Comment by danmanners** [comment]
Hey folks, I've been attempting to replicate everything since my account has been re-enabled to no avail. I've provisioned/de-provisioned things several times, and I'm finding nothing wrong so far. Going to keep attempting to re-create this.

**Comment by danmanners** [comment]
Hey folks, just a heads up I am now able to replicate this in Azure as well. Same steps, all of my Terraform code is here: https://github.com/danmanners/Azure-Terraform-Learning
